src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/CodeActions/Razor/ExtractToComponentCodeActionProvider.cs

```diff
@@ -83,7 +84,8 @@ public Task<ImmutableArray<RazorVSInternalCodeAction>> ProvideAsync(RazorCodeAct
 
     private static (SyntaxNode? Start, SyntaxNode? End) GetStartAndEndElements(RazorCodeActionContext context, RazorSyntaxTree syntaxTree)
     {
-        var owner = syntaxTree.Root.FindInnermostNode(context.StartAbsoluteIndex, includeWhitespace: true);
+        var startIndex = AdjustStartIndex(context.StartAbsoluteIndex);
```

Comment:
This has me thinking that one of the best ways to solve this might just be to reduce the selection to always be on a non-whitespace character. 

Move start forward and end back until we're on some kind of text....